x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/RBACEngine.java

```diff
@@ -880,19 +881,14 @@ static AuthorizedIndices resolveAuthorizedIndicesFromRole(
         }, name -> {
             final IndexAbstraction indexAbstraction = lookup.get(name);
             if (indexAbstraction == null) {
-                return false;
-            }
-            if (includeDataStreams) {
```

Comment:
Just to confirm that this is same to remove because core handles whether data stream is allowed in the context? Do we really need to remove it here? It does not seem to be strictly related to the original issue. It is also a bit inconsistent that `includeDataStreams` is still handled in the first lambda (the one backs `AuthorizedIndices.all`) but not here? Or should we also remove logic for `includeDataStreams` in the first lambda as well?

Reply:
> Just to confirm that this is same to remove because core handles whether data stream is allowed in the context? 

Yes. In Core, accessing a datastream in a context that doesn't allow datastreams is handled similarly to a missing index, or to an alias when the request options ignore aliases.

> Do we really need to remove it here? It does not seem to be strictly related to the original issue. 
On the contrary, I think it would be inconsistent to continue to deal with non-authz unavailable conditions.

> It is also a bit inconsistent that includeDataStreams is still handled in the first lambda (the one backs AuthorizedIndices.all) but not here? Or should we also remove logic for includeDataStreams in the first lambda as well?

Yes, I agree it looks inconsistent. But I don't think we can get away from it. When expanding wildcards we need to know if to expand to datastreams and/or aliases too, but for explicit names we don't (and shouldn't).

Reply:
We met today and also discussed this.
> It is also a bit inconsistent that includeDataStreams is still handled in the first lambda (the one backs AuthorizedIndices.all) but not here? Or should we also remove logic for includeDataStreams in the first lambda as well?

I think we're on the same page that it would be preferable to not check `includeDAtaStreams` when building the result of `AuthorizedIndices.all`. I think the code should work. But It's not obvious if it can affect performance. Therefore I prefer to keep this for a follow-up PR.